sigstore-java/src/main/java/dev/sigstore/KeylessSigner.java

```diff
@@ -76,40 +94,67 @@ public static class Builder {
     private RekorVerifier rekorVerifier;
     private OidcClient oidcClient;
     private Signer signer;
+    private Duration minSigningCertificateLifetime = Duration.ofMinutes(5);
 
+    @CanIgnoreReturnValue
     public Builder fulcioClient(FulcioClient fulcioClient, FulcioVerifier fulcioVerifier) {
       this.fulcioClient = fulcioClient;
       this.fulcioVerifier = fulcioVerifier;
       return this;
     }
 
+    @CanIgnoreReturnValue
     public Builder rekorClient(RekorClient rekorClient, RekorVerifier rekorVerifier) {
       this.rekorClient = rekorClient;
       this.rekorVerifier = rekorVerifier;
       return this;
     }
 
+    @CanIgnoreReturnValue
     public Builder oidcClient(OidcClient oidcClient) {
       this.oidcClient = oidcClient;
       return this;
     }
 
+    @CanIgnoreReturnValue
     public Builder signer(Signer signer) {
       this.signer = signer;
       return this;
     }
 
+    /**
+     * The signer will request a signing certificate from Fulcio if the current certificate is going
+     * to expire within this duration.
```

Comment:
Can this be a bit more descriptive?
```
The instance of the KeylessSigner will try to reuse a previously acquired certificate if the expiration time on the certificate is more than "minSigningCertificateLifetime" time away. Otherwise, it will make a new request (oidc/fulcio) to obtain a new updated certificate to use for signing. Default "minSigningCertificateLifetime is {@value #SOME_STATIC_FIELD}". See https://docs.sigstore.dev/fulcio/overview/ for fulcio certificate validity.
```